src/Raven.Studio/typescript/common/queryUtil.ts

```diff
@@ -35,7 +35,15 @@ class queryUtil {
         const escapedFieldName = queryUtil.escapeCollectionOrFieldName(fieldName);
         const escapedIndexName = queryUtil.escapeName(indexName);
         const escapedValueName = queryUtil.escapeName(value);
-        return `from index ${escapedIndexName} where ${escapedFieldName} == ${escapedValueName}`;
+        
+        const fromPart = `from index ${escapedIndexName}`;
+        let wherePart = `where ${escapedFieldName} == ${escapedValueName}`;
+        
+        if (indexName.startsWith("Auto") && (value.startsWith("{") || value === "NULL_VALUE")) {
```

Comment:
Better to do that by checking capital words, no?

Reply:
pls see this comment
https://issues.hibernatingrhinos.com/issue/RavenDB-13373/Automatic-queries-should-understand-casing#focus=Comments-67-385157.0-0


If the term is plain, i.e. `"Examples": "Test"` then it is indexed as lower-case (both "TEST" and "test" will be indexed as lower case "test")
and if you apply 'exact' you will not get all docs

Reply:
Try to run query like `from 'Employees' where exact(LastName = 'Callahan')` then the terms for field `exact(LastName)` created under the covers will not be also lowercased. So I agree with Oren that instead of checking if this is a JSON object we should determine that based on upper cased characters

Reply:
The underlying issue is this:

* `users/1` - `{"Name": "Oren"}`
* `users/2` - `{"Name": {"FirstName": "Oren"}}`

The query:

`from Users where Name = 'Oren'`

Now, go and look at the terms for this.
Because `Name` for `users/2` is an object, so we serialize as JSON, without tokenization

So we then go to the Terms, and click on the `{"FirstName": "Oren"}` term. Then we get a bad query, since we'll tokenize the query

Reply:
so the pr solves exactly that. 
when the term is an object we add the 'exact'.

Reply:
Regarding my comment - I thought that my example will be another case when this can not work correctly but I have checked that in the studio and everything is fine. The query that it generates is: `from index 'Auto/Employees/ByExact(LastName)' where 'exact(LastName)' == 'Buchanan'` which is okay since under the covers we do create additional field with name "exact(LastName)"

Reply:
Added now a commit that better detects if 'uppercase'
So pr adds `'exact'` only if both:
*  term is 'json' && 
*  has Upper